Project : This is a piece of software that is capable to open a CVS file or excel file. Change and manipulate pieces of data as imput to return a chain of features of an specific object.

In [13]:
#obsidian objects

obsidian_objects = "diverse artetefacts found in Nz from Pacific Communities"
print (obsidian_objects)

# add title to piece of software.

diverse artetefacts found in Nz from Pacific Communities


Sample type: is the main feature that include a list of objects  within a numerical order. This make an easy identification with the features in its row.

In [14]:
from pprint import *
#from intertools import * # to perform with efficient loops
#from networkx import * # to work with  graphics.
import csv

def read_my_file(filename):
    obsidian_artefacts = open(filename)
    # in order to open data that is previously in an accepted format, the variable must be equal to a comand OPEN that will get access to the data

    data = obsidian_artefacts.read()
    # this next part  will be able to read whate is inside my variable "obsidian_artefacts"

    datarows = data.split('\n')


    datarows
    # which will print my information without any speficific  instruction.



    #chop each string in datarows into a list of strings where each splitted string is a cell_value
    cell_values = []
    for eachrow in datarows:
        cell_values.append(eachrow.split(','))


    cell_values
    #print

    obs_dict = {}

    header_keys = cell_values[0]

    for col_index in range(0,len(header_keys)):

        key = header_keys[col_index]

        values=[]

        for row_index in range(1,len(cell_values)-1):
            eachrow = cell_values[row_index]
            values.append(eachrow[col_index])
# this will give me in average what is the quantity of data that is going to be manipulated.

        obs_dict[key] = values

    return obs_dict

    for row in readCSV:
            Sample_Type = row[0]
            Study = row[1]
            Area = row[2]
            Source_Specific = row[3]
            Sample_ID = row[4]
            XRF_Setting = row[5]
            Source_Zone = row[6]

            Sample_Types.append(Sample_Type)
            Studys.append(Study)
            Areas.append(Area)
            Source_Specifics.append(Source_Specific)
            Sample_IDs.append(Sample_ID)
            XRF_Settings.append(XRF_Setting)
            Source_Zones.append(Source_Zone)

            indexes = []
# in this part i classify the main features,per column.
            while True:
                confirmation = "n";
                userinput = input('Input command. Type help for list of commands: ')
        
# this is the key access Help as word will allow me to get options to continue

            if userinput == 'help':
                print("List of commands:")
                print("Study Area")
                print("Source specific")
                print("List Column (Type: List)")
                print("List Column names (Type: Keys)")
                continue
            #  the available imputs will be display and the options will be  any of the main features  sample type, study area or source specific.
                if userinput == "Sample_Type":
                    subcommand = input('Input Sample_Type:')
                    if confirmation == "y":
                        subindexes = []
                        for x in range (0, len(indexes)):
                            if (subcommand == Sample_Types[indexes[x]]):
                                subindexes.append(indexes[x])
                            indexes = []
                            indexes = subindexes
                    if confirmation == "n":
                        indexes = []
                        for x in range(0, len(Sample_Types)):
                            if subcommand == Sample_Types[x]:
                                indexes.append(x)
                if userinput == "Study Area":
                    subcommand = input('Input Study Area:')
                    if confirmation == "y":
                        subindexes = []
                        for x in range (0, len(indexes)):
                            if (subcommand == Studys[indexes[x]]):
                                subindexes.append(indexes[x])
                        indexes = []
                        indexes = subindexes
                    if confirmation == "n":
                        indexes = []
                        for x in range(0, len(Studys)):
                            if subcommand == Studys[x]:
                                indexes.append(x)
                if userinput == "Source specific":
                    subcommand = input('Input Source specific:')
                    if confirmation == "y":
                        subindexes = []
                        for x in range (0, len(indexes)):
                            if (subcommand == Source_Specifics[indexes[x]]):
                                subindexes.append(indexes[x])
                        indexes = []
                        indexes = subindexes
                    if confirmation == "n":
                        indexes = []
                        for x in range(0, len(Source_Specifics)):
                            if subcommand == Source_Specifics[x]:
                                indexes.append(x)
# This is a fundamental part of the code where for any number of the main list  the associate features  will be displayed as well
                if userinput == "List":
                    subcommand = input('Type in the column you wish to list: ')
                    subindexes = []
                    indexes = []
                    if (subcommand == "Sample_Type"):
                        for x in range(0, len(Sample_Types)):
                            print(Sample_Types[x])
                            continue
                    if (subcommand == "Study"):
                        for x in range(0, len(Studys)):
                            print(Studys[x])
                            continue
                if (subcommand == "Area"):
                    for x in range(0, len(Areas)):
                        print(Areas[x])
                        continue
                if (subcommand == "Source_Specific"):
                    for x in range(0, len(Source_Specifics)):
                        print(Source_Specifics[x])
                        continue
                if (subcommand == "Sample_ID"):
                    for x in range(0, len(Sample_IDs)):
                        print(Sample_IDs[x])
                        continue
                if (subcommand == "XRF_Setting"):
                    for x in range(0, len(XRF_Settings)):
                        print(XRF_Settings[x])
                        continue
                if (subcommand == "Source_Zone"):
                    for x in range(0, len(Source_Zones)):
                        print(Source_Zones[x])
                        continue
                    
# here we have a key words that will allow me to complete functions such as exit or any possible search
            if userinput == "Keys":
                print("The following keys exist: \n\n")
                print("Sample_Type\nStudy Area\nSource_Specific\nSample_ID\nXRF_Setting\nSource_Zone\nRb_ppm\nSr_ppm\nY_ppm\nZr_ppm\nNb_ppm\nSi_ppm\nTi_ppm\nAl_ppm\nFe_ppm\nMn_ppm\nMg_ppm\nCa_ppm\nNa_ppm\nK_ppm\nBag_Info")
                print("The following keys are searchable: \n\n")
                print("Sample_Type\nStudy Area\nSource_Specific\nSample_ID")
                continue

            if (len(indexes) == 0):
                print ("Nothing matches input specifications")
                continue
            
# in case that information will not be found then you have the option to continue 

            for y in range(0, len(indexes)):
                print ("Sample_Type: ",Sample_Types[indexes[y]])
                print ("Study Area: ",Studys[indexes[y]])
                print ("Source specific: ",Source_Specifics[indexes[y]])
                print ("Sample_ID: ",Sample_IDs[indexes[y]])
                print ("XRF_Setting: ",XRF_Settings[indexes[y]])
                print ("Source_Zone: ",Source_Zones[indexes[y]])
                print ("\n")
            confirmation = input('Do you wish to narrow the results? y/n: ')
            if confirmation == "n":
                indexes = []
            # this will give me the final option to reduce the quantity of the results available.


The main column is the pattern for any future improvement.

In [15]:
o_dict = read_my_file('obs-art.csv')
print('cols',len(o_dict.keys()))
print('rows',len(o_dict['Sample_Type']))

cols 21
rows 1070


The number or columns are 21 and rows 1070, It is important to remember that elements in the first column will give us the number of artefacts that we are going to use for to run this piece of software

In [16]:
o_dict.keys()

dict_keys(['K_ppm', 'Na_ppm', 'Source_Specific', 'Nb_ppm', 'Sample_ID', 'XRF_Setting', 'Bag_Info', 'Fe_ppm', 'Source_Zone', 'Al_ppm', 'Ca_ppm', 'Zr_ppm', 'Sample_Type', 'Rb_ppm', 'Mn_ppm', 'Ti_ppm', 'Y_ppm', 'Study Area', 'Sr_ppm', 'Mg_ppm', 'Si_ppm'])

We use [] with a list or dictionary to retrieve the data from it.
when we use it with list we need to give an index location inside the square braces e.g. list1[0]
when we use it with dict we need to give the key inside the square braces e.g. dict1['key1']

In [17]:
o_dict['Sample_Type']

['1Artefact',
 '2Artefact',
 '3Artefact',
 '4Artefact',
 '5Artefact',
 '6Artefact',
 '7Artefact',
 '8Artefact',
 '9Artefact',
 '10Artefact',
 '11Artefact',
 '12Artefact',
 '13Artefact',
 '14Artefact',
 '15Artefact',
 '16Artefact',
 '17Artefact',
 '18Artefact',
 '19Artefact',
 '20Artefact',
 '21Artefact',
 '22Artefact',
 '23Artefact',
 '24Artefact',
 '25Artefact',
 '26Artefact',
 '27Artefact',
 '28Artefact',
 '29Artefact',
 '30Artefact',
 '31Artefact',
 '32Artefact',
 '33Artefact',
 '34Artefact',
 '35Artefact',
 '36Artefact',
 '37Artefact',
 '38Artefact',
 '39Artefact',
 '40Artefact',
 '41Artefact',
 '42Artefact',
 '43Artefact',
 '44Artefact',
 '45Artefact',
 '46Artefact',
 '47Artefact',
 '48Artefact',
 '49Artefact',
 '50Artefact',
 '51Artefact',
 '52Artefact',
 '53Artefact',
 '54Artefact',
 '55Artefact',
 '56Artefact',
 '57Artefact',
 '58Artefact',
 '59Artefact',
 '60Artefact',
 '61Artefact',
 '62Artefact',
 '63Artefact',
 '64Artefact',
 '65Artefact',
 '66Artefact',
 '67Artefact',
 '68

In [18]:
one_art =[] # this is the  final conteiner where this will be storage.
for Sample_Type in o_dict: # 1  for each column  into the main " container" in this case o_dict  we will create something inside
    one_art.append(Sample_Type[0])# 3 we windicate this  will be added some where with append # 2 in each column the possition zero  has to be find